packages/techdocs-common/src/stages/generate/helpers.ts

```diff
@@ -286,6 +286,46 @@ export const patchMkdocsYmlPreBuild = async (
   }
 };
 
+/**
+ * Update docs/index.md file before TechDocs generator uses it to generate docs site,
+ * falling back to docs/README.md or README.md in case a default docs/index.md
+ * is not provided.
+ */
+export const patchIndexPreBuild = async ({
+  inputDir,
+  logger,
+}: {
+  inputDir: string;
+  logger: Logger;
+}) => {
+  const docsPath = path.join(inputDir, 'docs');
```

Comment:
A caveat, the `docs` could be something custom as well. :)
https://www.mkdocs.org/user-guide/configuration/#docs_dir

Reply:
+1, maybe read the `docs_dir` from `mkdocsYmlPath`?